sgrep/tests/js/deep_deep_stmt1.js

```diff
@@ -0,0 +1,71 @@
+
+function test1() {
+
+    //ERROR: match
+    var x = 1;
+
+    function foo() {
+        console.log(x);
+    }
+}
+
+function test2() {
+
+    //ERROR: match
+    var x = 1;
+
+    foo(() => {
+        console.log(x);
+    })
+}
+
+function test3() {
+
+    //ERROR: match
+    var x = 1;
+
+    foo(function() {
+        bar(() => {
+            console.log(x);
+        })
+    })
+}
+
+function test4() {
+
+    //ERROR: match
+    var x = 1;
+
+    class Hello {
+        
+	    test() {
+		    console.log(x);
+	    }
+        
+    }
+}
+
+function test5() {
+
+    //ERROR: match
+    var x = 1;
+
+    function OldSchool () {
+        
+        this.test = function() {
+            console.log(x);
+        }
+        
+    }
+}
+
```

Comment:
I'd recommend a negative test cases along the lines of:

```javascript
const foo = () => {
  var x = 1;
}
console.log("hi")
```

with a pattern like
```
var $X = 1;
...
console.log("...");
```